### PR TITLE
Controllers

### PR DESCRIPTION
**Pull request by kkmanuu**
✔️   For each URL I  create:

     ▶️  A route.
      ▶️  An action in the correct controller.
      ▶️ A view file.
✔️  I include only placeholders written in plain HTML

✔️  I Make sure that I  added code to handle only the 4 required URLs.

✔️ I Make sure that when I  run the rails routes command in your console

### CODE REVIEW

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @kkmanuu,

#### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

#### Your Code Review Request is marked as invalid for this reason:

- [ ] The collaborator name on my reviewer dashboard differs from the repo owner of the PR submitted. Please provide evidence that the same person owned both the Microverse student name which is `Mission Rono` and the Github repo owner name which is `Emmanuel Kipngeno`

image

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

**Comment by kkmanuu** [comment]
Screenshot (150)

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get '/', to: 'users#index', as: :users
+  get '/users/:id', to: 'users#show', as: :user
+  get '/users/:id/posts', to: 'posts#index', as: :user_posts
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: :user_post
 end
```
Comment (frank1738) [inline config/routes.rb:6]: - [Optional ] There's a concise and powerful way to define your routes using a more fine-grained approach, which can make your code cleaner and more explicit. Here is an example.

```

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end

```

**Review by frank1738 — APPROVED** [review]
Hi @kkmanuu , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Has a corresponding view file for each of the controller actions
✅ Has `users_controller` and `posts_controller` with `#index` and `#show` actions.👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**